python/ray/actor.py

```diff
@@ -397,10 +395,9 @@ def _remote(self,
             if actor_method_cpu == 1:
                 actor_placement_resources = resources.copy()
                 actor_placement_resources["CPU"] += 1
-
             function_signature = meta.method_signatures[function_name]
-            creation_args = signature.extend_args(function_signature, args,
-                                                  kwargs)
+            signature.validate_args(function_signature, args, kwargs)
```

Comment:
Seems like we do `validate_args` before `flatten_args` every time. Should either just combine these or call `validate_args` from within `flatten_args` to avoid bugs in the future if someone uses `flatten_args` elsewhere.